setup.py

```diff
@@ -106,7 +106,7 @@ def get_extensions():
     },
     
     install_requires = ["requests >= 2.12",
-                        "numpy >= 1.19",
+                        "numpy >= 1.19, <= 1.24",
```

Comment:
The upper version boundary is not necessary here, since more modern versions of NumPy should also work. The reason for the upper bound is `pyproject.toml`, is that a NumPy version as low as possible allows compilation of modules that work for that NumPy version and all versions above.

```suggestion
                        "numpy >= 1.19",
```